### PR TITLE
Fail gracefully if solc doesn't find any contracts. Information how t…

### PR DESCRIPTION
**Pull request by miohtama**
### What was wrong?

If solc did not find any contracs in given folder JSON output was not properly managed.

### How was it fixed?

Raise an exception if solc doesn't find any contracts.

Updates regarding solc installation instructions, .gitignore.

#### Cute Animal Picture

Deerp

### CODE REVIEW

File: tests/compilation/test_compile_empty.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    compile_files,
+)
+
+from solc.exceptions import ContractsNotFound
+
+
+def test_compile_empty_folder():
+    """Execute compile on a folder without contracts."""
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
```
Comment (pipermerriam) [inline tests/compilation/test_compile_empty.py]: I think this isn't python2 compatable.

You can do this with the `tmpdir` fixture provided by pytest.

```python
def test_things(tmpdir):
    dir_that_will_be_auto_cleaned_up = str(tmpdir.mkdir('some-dir'))
    ...  # do things.
```

**Comment by pipermerriam** [comment]
Also two linting errors.

```
/home/travis/build/pipermerriam/py-solc/solc/exceptions.py:10:55: W292 no newline at end of file
/home/travis/build/pipermerriam/py-solc/solc/main.py:46:8: E713 test for membership should be 'not in'
```

**Comment by pipermerriam** [comment]
Just the two small cleanup items and I'll get this merged and released.

**Comment by miohtama** [comment]
Done.

**Comment by pipermerriam** [comment]
Available as `0.3.0` on pypi and in the releases section.

**Pull request merged**